packages/jest-cli/src/TestRunner.js

```diff
@@ -67,18 +75,40 @@ class TestRunner {
     return this._buildPromise;
   }
 
-  _getAllTestPaths() {
-    return this._hasteMap
-      .matchFiles(this._testRegex)
-      .then(paths => paths.filter(path => this.isTestFilePath(path)));
+  _filterTestPathsWithStats(allPaths, options) {
+    const data = {
+      total: allPaths.length,
+      stats: {},
+    };
+
+    const testCases = options && options.checkTestPathPattern
+      ? Object.keys(this._testPathCases)
+      : Object.keys(this._testPathCases).slice(0, -1);
```

Comment:
Oh. I understand why you are doing the slice now. I'd prefer not to include `testPathPattern` in this object but instead add it separately if the option is provided:

```js
const testCases = Object.assign({}, this._testPathCases);
if (checkTestPathPattern) {
  testCases.pathPatterns = this._testPathPattern;
}
```

that seems safer :)